src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockWithEmoji.java

```diff
@@ -0,0 +1,52 @@
+/*
+EmptyBlock
+option = TEXT
+tokens = LITERAL_WHILE, LITERAL_TRY, LITERAL_CATCH, LITERAL_FINALLY, LITERAL_DO, LITERAL_IF, \
+         LITERAL_ELSE, LITERAL_FOR, INSTANCE_INIT, STATIC_INIT, LITERAL_SWITCH, LITERAL_DEFAULT, \
+         LITERAL_CASE
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.blocks.emptyblock;
+
+public class InputEmptyBlockWithEmoji {
+    // violation below 'Empty STATIC_INIT block'
+    static {
+
+    }
+    static {
+        String c = "🎄";
+    }
+    public void fooMethod()
+    {
+        String a = "12🤩";
+        String b = "";
+        if (a == "🤩12🧐🧐") { } // violation 'Empty if block'
```

Comment:
I might miss something, but what is point for this check to be converted to code points ?

unicode like symbols are possible in strings values, but Check does not care about values at all.

can you share output of CLI that shows that latest released checkstyle on this Input file gives false positives ?

Reply:
@romani You are correct since this check finds a curly token and checks for the empty line inside that token, codepoints are not needed.

I suspect there may be other checks as well which don't require changes.

Reply:
if we can reproduce false positive/negative by inputs - we do migration to codepoints. 
If we can not - we keep in PRs only update for tests and inputs. Maybe in future we will figure out a way, so we will extend tests. 

Your test update is good let's merge it no matter what.

Reply:
You can create path file at attach to this PR as file to keep it for future.

Or you can close this PR and create new PR for tests updates Only. In this case GitHub will store your code even PR become closed and not merged.

Reply:
<pre> java -jar ~/Desktop/checkstyle-10.1-SNAPSHOT-all.jar -c ~/Desktop/config.xml src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockWithEmoji.java
Starting audit...
[WARN] /Users/shabana/Documents/OpenSource/checkstyle/src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockWithEmoji.java:15:12: Empty STATIC_INIT block. [EmptyBlock]
[WARN] /Users/shabana/Documents/OpenSource/checkstyle/src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockWithEmoji.java:28:34: Empty if block. [EmptyBlock]
[WARN] /Users/shabana/Documents/OpenSource/checkstyle/src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockWithEmoji.java:30:62: Empty for block. [EmptyBlock]
[WARN] /Users/shabana/Documents/OpenSource/checkstyle/src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockWithEmoji.java:31:25: Empty if block. [EmptyBlock]
Audit done.
</pre>

Sorry, @romani I used the wrong config that's why not able to reproduce the error. The above output is when the config is

`<module name="EmptyBlockCheck">`
       `<property name="option" value="text"/>`
        `</module>`

Reply:
@MUzairS15 , please do full print as we use in issue description to show me all details.
Imagine you report a defect in Check, users need to know how to problem is reproducable.

Reply:
```
checkstyle % cat ~/Desktop/config.xml src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockWithEmoji.java  
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name = "Checker">
    <property name="charset" value="UTF-8"/>

    <property name="severity" value="warning"/>

    <property name="fileExtensions" value="java, properties, xml"/>

    <module name="TreeWalker">
        <module name="EmptyBlockCheck">
       <property name="option" value="text"/>
        </module>
    </module>
</module>

/*
EmptyBlock
option = TEXT
tokens = LITERAL_WHILE, LITERAL_TRY, LITERAL_CATCH, LITERAL_FINALLY, LITERAL_DO, LITERAL_IF, \
         LITERAL_ELSE, LITERAL_FOR, INSTANCE_INIT, STATIC_INIT, LITERAL_SWITCH, LITERAL_DEFAULT, \
         LITERAL_CASE


*/

package com.puppycrawl.tools.checkstyle.checks.blocks.emptyblock;

public class InputEmptyBlockWithEmoji {
    // violation below 'Empty STATIC_INIT block'
    static {

    }
    static {
        String c = "🎄";
    }
    public void fooMethod()
    {
        String a = "12🤩";
        String b = "";
        if (a == "🤩12🧐🧐") { } // violation 'Empty if block'
        char[] s = {'1', '2'};
        int index = 2;
        if (doSideEffect() == 1) { } // violation 'Empty if block'
        while ((a = "12") != "🧐") {return;} // ok
        for (; index < s.length && s[index] != 'x'; index++) {} // violation 'Empty for block'
        if (a == "12🤣") {} else {System.identityHashCode("a");} // violation 'Empty if block'
        // violation below 'Empty switch block'
        switch("😆😆😆😆😆") {

        }
        switch (a) {    // ok
            case "🎄": {
                a = "🤣🤣";
            }
            case "🙃":{} // violation 'Empty case block'
                a = "1223🤣";
            default:
                a = null;
        }
        switch(b) {case "🤩": break; default: { } } // violation 'Empty default block'
    }

    public int doSideEffect()
    {
        return 1;
    }
}
shabana@Shabanas-MacBook-Air checkstyle %  java -jar ~/Desktop/checkstyle-10.1-SNAPSHOT-all.jar -c ~/Desktop/config.xml src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockWithEmoji.java
Starting audit...
[WARN] /Users/shabana/Documents/OpenSource/checkstyle/src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockWithEmoji.java:15:12: Empty STATIC_INIT block. [EmptyBlock]
[WARN] /Users/shabana/Documents/OpenSource/checkstyle/src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockWithEmoji.java:28:34: Empty if block. [EmptyBlock]
[WARN] /Users/shabana/Documents/OpenSource/checkstyle/src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockWithEmoji.java:30:62: Empty for block. [EmptyBlock]
[WARN] /Users/shabana/Documents/OpenSource/checkstyle/src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/emptyblock/InputEmptyBlockWithEmoji.java:31:25: Empty if block. [EmptyBlock]
Audit done.

Reply:
@MUzairS15 please use latest release version of Checkstyle (not snapshot) and reduce input to the smallest failing case to show false negative.

Reply:
@romani  @nick-mancuso updated

Reply:
@romani https://github.com/checkstyle/checkstyle/pull/11366 CI is failing because check requires update. Let's close #11366 and continue here.

Reply:
thanks a lot for visual prove.